prboom2/src/g_game.c

```diff
@@ -3179,13 +3182,74 @@ void G_BeginRecording (void)
 {
   int i;
   byte *demostart, *demo_p;
+  int num_extensions = 0;
   demostart = demo_p = malloc(1000);
   longtics = 0;
 
-  if (umapinfo_loaded)
+  // ano - jun2019 - add the extension format if needed
+  if (gamemapinfo)
+  {
+    num_extensions++;
+  }
+
+  if (num_extensions > 0)
   {
-	  *demo_p++ = 255;
+    // demover
+    *demo_p++ = 0xFF;
+    // signature
+    *demo_p++ = prdemosig[0]; // 'P'
+    *demo_p++ = prdemosig[1]; // 'R'
+    *demo_p++ = prdemosig[2]; // '+'
+    *demo_p++ = prdemosig[3]; // 'U'
+    *demo_p++ = prdemosig[4]; // 'M'
+    *demo_p++ = '\0';
```

Comment:
Six bytes for a signature? How many ports are we expecting to support this? 
Also, why do we need a terminating NULL byte? This isn't a proper string, anyway (you don't even use `strcat()` et al here), but just a couple of ASCII bytes embedded into some binary cruft. I'd prefer if the signature would be four bytes, e.g. "PR+U".

Reply:
This, Eternity, and MBF all do this. Six bytes. The intent is that EE can read these too.

Reply:
Ah, indeed, even MBF has such a weird signature in the header.

Reply:
Actually, I think the MBF signature prboom carries is still left there, so there are actually 2 signatures in the demo, which I dont think makes much sense:
https://github.com/XaserAcheron/prboom-plus/blob/45b9e0806285d485ff9e1ba7757381021e7f9b24/prboom2/src/g_game.c#L3273-L3279

Reply:
It's layered. The umapinfo data wraps the normal demo, and that following signature determines the compatibility. It could be mbf, but it could also be boom (which has a different signature), etc. This is my understanding anyway. If eternity doesn't include the mbf signature, how will the demo compatibility be detected? It's not assumed that all umapinfo wads are mbf, I assume.

Reply:
Could this question get a bit more elaboration, please? Especially the parts that @Ferk and @kraflab asked are also still unclear to me.